routers/web/repo/commit.go

```diff
@@ -458,3 +464,54 @@ func processGitCommits(ctx *context.Context, gitCommits []*git.Commit) ([]*git_m
 	}
 	return commits, nil
 }
+
+// GroupedCommits defines the structure for grouped commits.
+type GroupedCommits struct {
+	Date    timeutil.TimeStamp
+	Commits []*git_model.SignCommitWithStatuses
+}
+
+// GroupCommitsByDate groups the commits by date (in days).
+func GroupCommitsByDate(commits []*git_model.SignCommitWithStatuses) []GroupedCommits {
+	// Use Unix timestamp of date as key (truncated to day)
+	grouped := make(map[int64][]*git_model.SignCommitWithStatuses)
+
+	for _, commit := range commits {
+		var sigTime time.Time
+		if commit.Committer != nil {
+			sigTime = commit.Committer.When
+		} else if commit.Author != nil {
+			sigTime = commit.Author.When
+		}
+
+		// Truncate time to date part (remove hours, minutes, seconds)
+		year, month, day := sigTime.Date()
+		dateOnly := time.Date(year, month, day, 0, 0, 0, 0, sigTime.Location())
```

Comment:
I do not think this algorithm is right.

For example: 
* commit1 is at "2025-04-10T08:00:00+08:00"
* commit2 is at "2025-04-09T23:00:00-01:00"
Your algorithm treats them as 2 different days.

Actually they are the same time, and the day depends on visitor's timezone.
* When visitor's timezone is `UTC`, time is `2025-04-10T00:00:00Z`, on local day "2025-04-10"
* When visitor's timezone is `-04:00`, time is `2025-04-09T20:00:00-04:00`, on local day "2025-04-09"

ps: I have no idea how GitHub handles such timezone problem. Just FYI.

Reply:
Now switched to using UTC for grouping, and corresponding test cases have also been added.